pkg/controller/installation/core_controller_test.go

```diff
@@ -1010,6 +1010,23 @@ var _ = Describe("Testing core-controller installation", func() {
 			Expect(*fc.Spec.BPFEnabled).To(BeTrue())
 		})
 
+		It("should set BPFEnabled to true on FelixConfiguration on a fresh install in BPF Mode", func() {
+			network := operator.LinuxDataplaneBPF
+			fc := &crdv1.FelixConfiguration{}
+			logr := log.WithValues()
+			cr.Spec.CalicoNetwork = &operator.CalicoNetworkSpec{LinuxDataplane: &network}
+			cr.Spec.CNI = &operator.CNISpec{}
+			updated, err := r.setDefaultsOnFelixConfiguration(ctx, cr, fc, logr)
```

Comment:
Like the tests below this test should call Reconcile instead of this function directly